pyo3-derive-backend/src/pymethod.rs

```diff
@@ -31,28 +31,43 @@ pub fn gen_py_method(
             impl_py_method_def_static(name, doc, &impl_wrap_static(cls, name, &spec))
         }
         FnType::Getter(ref getter) => {
-            impl_py_getter_def(name, doc, getter, &impl_wrap_getter(cls, name))
+            let mut takes_py = false;
+            for arg in &spec.args {
```

Comment:
Could this fail with a function definition like `foo(&self, py: Python, py: Python)`? In that case a slice pattern might be better.

Reply:
Slice pattern works nicely here, thanks :+1: